### PR TITLE
STICKY: GCC 6

### PR DESCRIPTION
**Pull request by alarcher**
Tracking progress on GCC 7-built userland.

### CODE REVIEW

File: components/network/isc-dhcp/Makefile [hunk 1]
```diff
@@ -23,28 +23,32 @@
 include ../../../make-rules/shared-macros.mk
 
 COMPONENT_NAME=		isc-dhcp
-COMPONENT_VERSION=	4.3.4
-IPS_COMPONENT_VERSION=	4.3.4
+COMPONENT_VERSION=	4.3.5
```
Comment (pyhalov) [inline components/network/isc-dhcp/Makefile]: Let's merge this update. (I hope, it doesn't break 4.9 build)

File: components/library/dbus/Makefile [hunk 2]
```diff
@@ -16,29 +16,31 @@
 include ../../../make-rules/shared-macros.mk
 
 COMPONENT_NAME= dbus
-COMPONENT_VERSION= 1.10.12
+COMPONENT_VERSION= 1.10.18
```
Comment (pyhalov) [inline components/library/dbus/Makefile]: Should we merge this (dbus update)? There's already 1.10.20

**Comment by alarcher** [comment]
@pyhalov @AndWac @xen0l More fixes to cherry-pick...

File: components/developer/luajit/patches/luajit-01-Makefile.patch [hunk 3]
```diff
@@ -14,3 +14,13 @@
  INSTALL_LIB=   $(DPREFIX)/$(MULTILIB)
  INSTALL_SHARE= $(DPREFIX)/share
  INSTALL_INC=   $(DPREFIX)/include/luajit-$(MAJVER).$(MINVER)
+318c318,324
```
Comment (pyhalov) [inline components/developer/luajit/patches/luajit-01-Makefile.patch]: Please, fix it to be in unified format

File: components/fix-stuff.mk [hunk 4]
```diff
@@ -0,0 +1,30 @@
+USERLAND_COMPONENTS_DIR ?= .
```
Comment (pyhalov) [inline components/fix-stuff.mk]: Please, don't add this file to oi-userland

File: components/openindiana/illumos-gate/Makefile [hunk 5]
```diff
@@ -19,18 +19,18 @@ COMPONENT_NAME=		illumos-gate
 COMPONENT_SRC=		$(COMPONENT_NAME)
```
Comment (pyhalov) [inline components/openindiana/illumos-gate/Makefile]: Avoid adding this change to oi-userland

File: make-rules/ips-buildinfo.mk [hunk 6]
```diff
@@ -36,7 +36,7 @@ RELEASE_MAJOR ?= 2017
 #
 # Release minor number: 0, 1, 2, etc.
 #
-RELEASE_MINOR ?= 0
+RELEASE_MINOR ?= 1
```
Comment (pyhalov) [inline make-rules/ips-buildinfo.mk]: @@Don't bump RELEASE_MINOR, better update RELEASE_MAJOR

File: components/Makefile [hunk 7]
```diff
@@ -155,3 +155,31 @@ $(COMPONENT_DIRS.nosetup):	$(WS_LOGS) FORCE
 	 $(BASS_O_MATIC) --make $(TARGET) $(LOG)
 
 FORCE:
+
+list-unpublished:
```
Comment (pyhalov) [inline components/Makefile]: Do we need it in oi-userland?

**Comment by alarcher** [comment]
@pyhalov I'll remove the commits that are not needed when we decide to merge. Until then since I build on two machines I need to have a branch with the stuff I need.

File: components/database/mongodb-33/Makefile [hunk 8]
```diff
@@ -77,12 +77,16 @@ install:	$(INSTALL_64) $(INSTALL_TOOLS)
 
 test:		$(NO_TESTS)
 
-REQUIRED_PACKAGES += SUNWcs
+# Build dependencies
 REQUIRED_PACKAGES += developer/build/scons
-REQUIRED_PACKAGES += developer/gcc-7
```
Comment (pyhalov) [inline components/database/mongodb-33/Makefile]: Please, preserve developer/gcc-7 build dependency

File: components/database/mongodb-34/Makefile [hunk 9]
```diff
@@ -107,13 +107,17 @@ install:	$(INSTALL_64) $(INSTALL_TOOLS)
 
 test:		$(TEST_64)
 
-REQUIRED_PACKAGES += SUNWcs
+# Build dependencies
 REQUIRED_PACKAGES += developer/build/scons
-REQUIRED_PACKAGES += developer/gcc-7
 REQUIRED_PACKAGES += developer/golang
```
Comment (pyhalov) [inline components/database/mongodb-34/Makefile]: please, preserve developer/gcc-7 build dependency

**Comment by pyhalov** [comment]
I've checked that after Ruby fixes augeas ugly hack is not necessary

**Pull request merged**